source/frontend/src/features/mapSideBar/lease/tabs/consultations/edit/ConsultationEditForm.tsx

```diff
@@ -0,0 +1,158 @@
+import { Formik, FormikHelpers } from 'formik';
+import styled from 'styled-components';
+
+import { FastDatePicker, Input, Select, TextArea } from '@/components/common/form';
+import { ContactInputContainer } from '@/components/common/form/ContactInput/ContactInputContainer';
+import ContactInputView from '@/components/common/form/ContactInput/ContactInputView';
+import { PrimaryContactSelector } from '@/components/common/form/PrimaryContactSelector/PrimaryContactSelector';
+import { YesNoSelect } from '@/components/common/form/YesNoSelect';
+import LoadingBackdrop from '@/components/common/LoadingBackdrop';
+import { Section } from '@/components/common/Section/Section';
+import { SectionField } from '@/components/common/Section/SectionField';
+import { StyledDivider } from '@/components/common/styles';
+import * as API from '@/constants/API';
+import SidebarFooter from '@/features/mapSideBar/shared/SidebarFooter';
+import { StyledFormWrapper } from '@/features/mapSideBar/shared/styles';
+import useLookupCodeHelpers from '@/hooks/useLookupCodeHelpers';
+import { getCancelModalProps, useModalContext } from '@/hooks/useModalContext';
+import { isOrganizationResult } from '@/interfaces';
+import { exists, isValidId } from '@/utils';
+
+import { UpdateConsultationYupSchema } from './EditConsultationYupSchema';
+import { ConsultationFormModel } from './models';
+
+export interface IConsultationEditFormProps {
+  isLoading: boolean;
+  initialValues: ConsultationFormModel | null;
+  onSubmit: (
+    values: ConsultationFormModel,
+    formikHelpers: FormikHelpers<ConsultationFormModel>,
+  ) => Promise<void>;
+  onCancel: () => void;
+}
+
+export const ConsultationEditForm: React.FunctionComponent<IConsultationEditFormProps> = ({
+  isLoading,
+  initialValues,
+  onSubmit,
+  onCancel,
+}) => {
+  const { setModalContent, setDisplayModal } = useModalContext();
+
+  const { getOptionsByType } = useLookupCodeHelpers();
+  const consultationTypeCodes = getOptionsByType(API.CONSULTATION_TYPES);
+
+  const cancelFunc = (resetForm: () => void, dirty: boolean) => {
+    if (!dirty) {
+      resetForm();
+      onCancel();
+    } else {
+      setModalContent({
+        ...getCancelModalProps(),
+        handleOk: () => {
+          resetForm();
+          setDisplayModal(false);
+          onCancel();
+        },
+      });
+      setDisplayModal(true);
+    }
+  };
+
+  const headerTitle = !isValidId(initialValues.id) ? 'Add Consultation' : 'Update Consultation';
+
+  return (
+    initialValues && (
+      <StyledFormWrapper>
+        <Formik<ConsultationFormModel>
```

Comment:
I think these forms could use the <UnsavedChangesPrompt /> component as well.

Reply:
Not sure what you mean here.

Reply:
Oh, one of my comments must not have been saved, referring to the ConfirmNavigation component.